window/src/os/x11/window.rs

```diff
@@ -1754,15 +1801,23 @@ impl WindowOps for XWindow {
 
     fn set_inner_size(&self, width: usize, height: usize) {
         XConnection::with_window_inner(self.0, move |inner| {
-            inner
-                .conn()
-                .send_request_no_reply_log(&xcb::x::ConfigureWindow {
+            let conn = inner.conn();
+            // XXX: Can I call eglWaitClient() or ensure that it has already been called by now?
```

Comment:
I'm not sure about this specific question, but if this window is in GL mode, rather than webgpu mode, then https://docs.rs/glium/latest/glium/struct.SyncFence.html#method.wait is the primitive that should be used, so presumably you'd want to do something with that. I'm unclear exactly what set of operations are being awaited and whether it is actually valid to wait on the fence here

Reply:
Oof, I've never used sync fences before, but they look like they could work.  All I would really like to do is wait until OpenGL is done drawing and swapping buffers so that I can safely resize the window.  `eglWaitClient()` has the nice semantics of guaranteeing exactly that, although if I can call `glFinish()` then that should also work.

And webgpu... I don't know what the equivalents of any of these things would be in that, but I imagine that there must be an equivalent of at least one of them... :thinking: